### ISSUE TITLE
github desktop is not syncing upstream state reliably

### ISSUE THREAD

**Issue by Kherae** [desktop/desktop#11312]
As of a somewhat recent update, github desktop stopped properly updating upstream states. This includes pull requests and any commits from a master branch. In my case, I am on the master branch of a fork of a master branch.

merging into current branch or pulling from origin do not function until after using both and restarting the client, and the client only recognizes  a completed pull request after that. Until those steps are done, the client does not see any commits from the master branch either.
this behavior is being exhibited on both my laptop and my desktop both, though they are both on the same operating system.
the branch in question is a  fork of a master copy.

Github desktop version 2.6.1
windows 10 home, 10.0.19041 build 19041

**Comment by steveward** [comment]
Thanks for the report @Kherae. Are updates to the `origin` remote being reflected correctly in GitHub Desktop? Can you confirm if you have the option to perform background fetching enabled? You can check this by going to `File` > `Options` > `Advanced`.

It would also be helpful if you could upload a recent log file from a day this issue occurred. To access the log files go to the file menu in GitHub Desktop and select `Help` > `Show Logs`.

**Comment by Kherae** [comment]
background fetching is on.

didnt see anything in any of the logs apart from a conflict I resolved, and i'm not sure which day had it so i'll attach the ones from that rough time period.
[logs.zip](https://github.com/desktop/desktop/files/5766583/logs.zip)

i'm also not sure about some git terminology.  what is 'origin remote'? if that means the original repository that my fork is...forked...from, then no, updates arent being reflected properly. this issue usually only happens when I had a pull request open that was merged, though.

**Comment by steveward** [comment]
The `origin` remote is just the remote repository that is connected to your local repository. The `upstream` would be the parent repository of your `origin` in the case of a fork. This sounds similar to what was just reported in https://github.com/desktop/desktop/issues/11334, but I've been unable to reproduce this as of yet.

**Comment by Kherae** [comment]
well, in that case i'll try to provide more detail, see if we can't find some common causes?

the repo in question is this one: https://github.com/Kherae/FrackinUniverse/
some things which have changed since maybe around the same time (i'm not reliable for info on this, due to memory problems), the upstream has some special things (idk the term) which check every pull request. 'Travis CI' and 'Github Actions'

Trying to think this through from a troubleshooting standpoint:
Github desktop is not properly fetching origin remote state, for one reason or another.
This would of course require verifying that origin remote and upstream are in sync, yes? If they -are- then that means that the program itself is not fetching data from either.  Which could mean anything from an internal version counter being set incorrectly to it simply not receiving data from origin about the state of the repository. Or it is receiving it and just...not acting on it.
given the steps I detailed, maybe it's some form of desync in the code?

I'm keeping an eye out for a repeat of the issue to see if I can get more accurate detail, hopefully I can find something that helps.

**Comment by raineorshine** [comment]
When I hit Fetch, pull requests are not updated and only the following is added to the logs:

```
2021-01-07T16:06:31.997Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: em - raineorshine (has token)
```

When I switch my Current Repository and then switch back, they are updated. The following is added to the logs:

```
2021-01-07T16:11:45.004Z - info: [ui] [BranchPruner] Pruning 0 branches that have been merged into the default branch, master (6639d2a30723e8a11ecea20bb1967a589140b7bc), from 'em-typedoc-theme
2021-01-07T16:11:50.723Z - info: [ui] Executing getRecentBranches: git log -g --no-abbrev-commit --pretty=oneline HEAD -n 2500 -- (took 1.069s)
2021-01-07T16:11:54.676Z - info: [ui] [BranchPruner] Last prune took place in 4 hours - skipping
```

If there are old OSX binaries I can try to find the version that broke. I tried building the project from source but it failed.

**Comment by steveward** [comment]
It looks like you may be running into this issue:

https://github.com/desktop/desktop/issues/11334#issuecomment-756627312

Please feel free to try out the beta version to see if that resolves the issue for you. Let me know if that does not fix things and we can investigate further.
